pubsub/subscriptions/pull_exactly_once_delivery.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_subscriber_exactly_once]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// receiveMessagesWithExactlyOnceDeliveryEnabled instantiates a subscriber client.
+// This differs from regular subscribing since you must call msg.AckWithResult() or msg.NackWithResult()
+// instead of the regular Ack/Nack methods. When exactly once delivery is enabled on the subscription,
+// the message is guaranteed to not be delivered again if the ack result succeeds.
+func receiveMessagesWithExactlyOnceDeliveryEnabled(w io.Writer, projectID, subID string) error {
+	// projectID := "my-project-id"
+	// subID := "my-sub"
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	sub := client.Subscription(subID)
+
+	// Receive messages for 10 seconds, which simplifies testing.
+	// Comment this out in production, since `Receive` should
+	// be used as a long running operation.
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	err = sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+		fmt.Fprintf(w, "Got message: %q\n", string(msg.Data))
+		r := msg.AckWithResult()
```

Comment:
Can we add a comment similar to the publish sample: https://github.com/GoogleCloudPlatform/golang-samples/blob/83461c3e2568fd4db5bc5717d9eb31b63ab33170/pubsub/topics/publish.go#L41. Basically referring to the blocking behavior here.